server/task/src/main/scala/org/apache/james/task/eventsourcing/EventSourcingTaskManager.scala

```diff
@@ -91,19 +95,25 @@ class EventSourcingTaskManager @Inject @VisibleForTesting private[eventsourcing]
     eventSourcingSystem.dispatch(command)
   }
 
-  override final def await(id: TaskId): TaskExecutionDetails = {
-    val details = getExecutionDetails(id)
-    if (details.getStatus.isFinished) {
-      details
-    } else {
-      Flux.from(terminationSubscriber.listenEvents)
-          .filter{
+  override final def await(id: TaskId, timeout: Duration): AwaitedTaskExecutionDetails = {
```

Comment:
don't you think `Optional<Duration>` could be better?

Reply:
I think the `submit()` need to support timeout too

Reply:
No, because you would not not if you had an `Option.empty` due  to an `TaskNotFound` or a `TimeOut`.

You want it to fail if you can not submit a Task.